sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -158,30 +158,37 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
 
     // For CREATE TABLE [AS SELECT], we should use the v1 command if the catalog is resolved to the
     // session catalog and the table provider is not v2.
-    case c @ CreateTable(ResolvedV1Identifier(ident), _, _, _, _) =>
+    case c @ CreateTable(ResolvedV1Identifier(ident), _, _, u: UnresolvedTableSpec, _,
```

Comment:
QQ: why separate UnresolvedOptionsList from UnresolvedTableSpec? Now the rule has to match both UnresolvedTableSpec and UnresolvedOptionsList.

Reply:
Nit: shall we have a single rule for resolving the table spec? This rule is for resolving session catalogs.

Reply:
Sure, I switched it back to using `ResolveTableSpec` as a separate rule.